### PR TITLE
[DO_NOT_MERGE] 6.5.1 release

### PR DESCRIPTION
**Pull request by palrahul**
PR to Preview what will be send as an official 6.5.1 PR to MoPub

We **do not intent to merge this**, as we would actually create PR on upstream repo.
And when it gets approve, we would pull it from upstream and merge into our repo's master

### CODE REVIEW

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 1]
```diff
@@ -38,20 +43,24 @@
...
     private static final String PLACEMENT_ID_KEY = "pid";
     private static final String PLACEMENT_IDS_KEY = "pids";
-    
+    private static final String KEY_AD_HEIGHT = "com_mopub_ad_height";
+    private static final String KEY_AD_WIDTH = "com_mopub_ad_width";
+
     private CustomEventBannerListener mCustomEventBannerListener;
     private final Handler mHandler;
     private String mAppId;
     private static String mPlacementId;
```
Comment (alexander-gubov) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: why mPlacementId is static? 2 Banners can have different placementIds

Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: Yes, I'm also have this question. MoPub committed this code: https://github.com/mopub/mopub-android-mediation/commit/566d53ea3ae1b09c5d5e9fe483b19a5bcfdf70f3#diff-87bfb1fb1eb3bd3c6c726f1d94fc22f1R45

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 2]
```diff
@@ -38,20 +43,24 @@
     private static final String APP_ID_KEY = "appId";
     private static final String PLACEMENT_ID_KEY = "pid";
     private static final String PLACEMENT_IDS_KEY = "pids";
-    
+    private static final String KEY_AD_HEIGHT = "com_mopub_ad_height";
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:46]: MoPub removed these two lines and replaced with `com.mopub.common.DataKeys.AD_WIDTH` and `com.mopub.common.DataKeys.AD_HEIGHT`.

File: Vungle/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,9 @@
 ## Changelog
+
+  *  6.5.1.0
```
Comment (alexander-gubov) [inline Vungle/CHANGELOG.md:3]: shouldn't we update our .aar in /libs as well?

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 4]
```diff
@@ -38,20 +43,24 @@
...
-    private VungleNativeAd vungleBannerAd;
+    private com.vungle.warren.VungleBanner vungleBannerAd;
+    private VungleNativeAd vungleMrecAd;
     private Context mContext;
     @NonNull
     private VungleAdapterConfiguration mVungleAdapterConfiguration;
     private AtomicBoolean pendingRequestBanner = new AtomicBoolean(false);
     private AdConfig adConfig = new AdConfig();
+
     private int adWidthInDp, adHeightInDp;
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: I think these two fields can be converted to local variables.

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 5]
```diff
@@ -122,22 +129,44 @@ public void run() {
...
+
+        if (AdSize.isBannerAdSize(vungleAdSize)) {
+            if (sVungleRouter.isBannerAdPlayable(mPlacementId, vungleAdSize)) {
+                mVungleRouterListener.onAdAvailabilityUpdate(mPlacementId, true);
+                MoPubLog.log(mPlacementId, LOAD_SUCCESS, ADAPTER_NAME);
+            } else {
+                sVungleRouter.loadBannerAd(mPlacementId, vungleAdSize, mVungleRouterListener);
+                MoPubLog.log(mPlacementId, LOAD_ATTEMPTED, ADAPTER_NAME);
+            }
+        } else if (VUNGLE_MREC == getVungleAdSize(localExtras)) {
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: Here we don't need to fire `getVungleAdSize()` again. We can use `vungleAdSize` variable directly.

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 6]
```diff
@@ -122,22 +129,44 @@ public void run() {
...
-            sVungleRouter.loadAdForPlacement(mPlacementId, mVungleRouterListener);
-            MoPubLog.log(mPlacementId, LOAD_ATTEMPTED, ADAPTER_NAME);
         }
     }
 
     private AdConfig.AdSize getVungleAdSize(Map<String, Object> localExtras) {
+        AdConfig.AdSize adSizeType = null;
         Preconditions.checkNotNull(localExtras);
 
         final Object adWidthObject = localExtras.get(AD_WIDTH);
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: int adWidthInDp = 0;

File: build.gradle [hunk 7]
```diff
@@ -29,6 +29,7 @@ subprojects {
         maven { url "https://adcolony.bintray.com/AdColony" }
         maven { url 'https://dl.bintray.com/ironsource-mobile/android-sdk'}
         maven { url 'https://tapjoy.bintray.com/maven' }
+        maven { url "https://jitpack.io" }
```
Comment (HuFangshuai) [inline build.gradle]: small suggestion: I know we use jitpack for RC release but MoPub don't use this maven to fetch our aar. So I think we can remove this one.

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 1 above)
Comment (palrahul) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: Not sure why they added it as static, have removed it

File: Vungle/CHANGELOG.md (hunk 3 above)
Comment (palrahul) [reply Vungle/CHANGELOG.md:3]: Not sure about MoPub's policy, I do see some AdNetworks adding it in their PR, if after upstream PR submission there would be additional changes during PR review the .aar could be outdated, unless we or MoPub later update it in event of any further changes.

**Pull request closed without merging**